app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,31 @@
+class CommentsController < ApplicationController
+  before action :set_post, except: [:new]
+
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment =
+      @post.comments.build(comment_params.merge(user:
+      @current_user))
+
+    if @comment.save
+      redirect_to user_post_path(@post.author, @post),
+                  notice: 'Comment added!'
+    else
+      render 'posts/show', alert: 'Unable to add
+            comment.'
+    end
+  end
+
```

Comment:
- [ ] Kindly fix the argument error preventing the user from creating a comment for a post.

![image](https://github.com/IvonneBenitesRodriguez/rails_blog/assets/81932478/3f2b22d1-1216-4c47-8585-baa820a9d9b4)

Video 
[cinnamon-2023-12-16T000124+0200.webm](https://github.com/IvonneBenitesRodriguez/rails_blog/assets/81932478/e267ad00-5370-4ff3-92c1-7811350f81b6)

Reply:
Dear @Munsa1 
thank you so much for your kind suggestion and explanations,
I appreciate it. I modified the changes you suggested me.
Thanks again, 
Best wishes,